model/dependencies.go

```diff
@@ -90,8 +99,24 @@ func (di *dependencyIncluder) handle(pair TVPair) (bool, error) {
 
 	// queue up all dependencies for recursive inclusion
 	deps := di.expandDependencies(pair, bvt.DependsOn)
+
+	// If this function is invoked from generate.tasks, calculate all variants
+	// that need to be included as dependencies, but are not in the generated project.
+	// If all the task / variant pairs that spawn these dependencies are inactive, we
+	// also mark this newly generated variant as inactive.
+	pairSpecifiesActivation := activationInfo != nil && activationInfo.taskOrVariantHasSpecificActivation(pair.Variant, pair.TaskName)
 	for _, dep := range deps {
-		ok, err := di.handle(dep)
+		if activationInfo != nil && !activationInfo.variantExistsInGeneratedProject(dep.Variant) {
```

Comment:
It seems like `variantExistsInGeneratedProject` is the only usage of `generatedProjectVariants`, and that field is not actually needed by `specificActivationInfo` once it's exited from `IncludeDependencies`. Can we just pass `g.BuildVariants` as a separate parameter into `IncludeDependencies` instead of passing it in with the `specificActivationInfo`? Might save some confusion on the purpose of that field, since it's only used in one specific function.

Reply:
I considered this but went with the current approach for fear of overloading `IncludeDependencies` and its helpers with parameters, but it does seem a little needless to add a new field to `specificActivationInfo` that is sort of unrelated to its purpose. Done.